engines/agi/menu.cpp

```diff
@@ -82,26 +85,34 @@ void GfxMenu::addMenu(const char *menuText) {
 
 	menuEntry->textLen = menuEntry->text.size();
 
-	// Cut menu name in case menu bar is full
-	// Happens in at least the fan game Get Outta Space Quest
-	// Original interpreter had graphical issues in this case
-	// TODO: this whole code needs to get reworked anyway to support different types of menu bars depending on platform
-	curColumnEnd += menuEntry->textLen;
-	while ((menuEntry->textLen) && (curColumnEnd > 40)) {
-		menuEntry->text.deleteLastChar();
-		menuEntry->textLen--;
-		curColumnEnd--;
+	if (!_vm->isLanguageRTL()) {
+		// Cut menu name in case menu bar is full
+		// Happens in at least the fan game Get Outta Space Quest
+		// Original interpreter had graphical issues in this case
+		// TODO: this whole code needs to get reworked anyway to support different types of menu bars depending on platform
+		curColumnEnd += menuEntry->textLen;
+		while ((menuEntry->textLen) && (curColumnEnd > 40)) {
+			menuEntry->text.deleteLastChar();
+			menuEntry->textLen--;
+			curColumnEnd--;
+		}
 	}
 
 	menuEntry->row = 0;
-	menuEntry->column = _setupMenuColumn;
+	if (!_vm->isLanguageRTL()) 
+		menuEntry->column = _setupMenuColumn;
+	else
+		menuEntry->column = _setupMenuColumn - menuEntry->textLen;
 	menuEntry->itemCount = 0;
 	menuEntry->firstItemNr = _itemArray.size();
 	menuEntry->selectedItemNr = menuEntry->firstItemNr;
 	menuEntry->maxItemTextLen = 0;
 	_array.push_back(menuEntry);
 
-	_setupMenuColumn += menuEntry->textLen + 1;
+	if (!_vm->isLanguageRTL())
+		_setupMenuColumn += menuEntry->textLen + 1;
+	else
+		_setupMenuColumn -= menuEntry->textLen + 1;
```

Comment:
Can be: `_setupMenuColumn += _vm->languageDirection() * (menuEntry->textLen + 1)`